lib: nrf_modem: Clear the address structure on recvfrom (#6545)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

Modem libraries
---------------

* :ref:`nrf_modem_lib_readme` library:

>   * Fixed a bug in the socket offloading component, where the :c:func:`recvfrom` wrapper could do an out-of-bounds copy of the sender address when the application was compiled without IPv6 support. In some circumstances, the out of bounds copy could result in :c:func:`send` and other socket API calls to block indefinitely.

Reviewer: ```suggestion
  * Fixed a bug in the socket offloading component, where the :c:func:`recvfrom` wrapper could do an out-of-bounds copy of the sender's address, when the application is compiled without IPv6 support. In some cases, the out of bounds copy could indefinitely block the :c:func:`send` and other socket API calls.
```
